### ISSUE TITLE
Docker for Win acme.json permissions problem.

### ISSUE THREAD

**Issue by JaronrH** [containous/traefik#1807]
**Traefik v1.3.1
Docker CE for Windows: 17.06.0-ce-win18 (12627)**

I have the /acme folder routed to a host volume which contains the file acme.json.  With the Traefik 1.3.1 update, I noticed that Traefik gets stuck in an infinite loop complaining that the "permissions 755 for /etc/traefik/acme/acme.json are too open, please use 600".  The only solution I've found is to remove acme.json and let Traefik re-negotiate the certs.  Unfortunately, if I need to restart the container, I have to remove acme.json again or I'm stuck with the same issue again!

My guess is that the issue lies with the Windows volume mapped to Docker but I was wondering what the recommended workaround would even be for this?

**Comment by ldez** [comment]
Thanks for your interest in Traefik !

Every question in the issues means features less 😉.

The issue tracker is for reporting bugs and feature requests only.
For end-user related support questions, refer to one of the following:

- [Stack Overflow](https://stackoverflow.com/questions/tagged/traefik) using the "traefik" tag
- the [Traefik community Slack channel](https://traefik.herokuapp.com)

**Comment by eosrei** [comment]
This Github issue comes up on Google before the currently-unanswered [StackOverflow version](https://stackoverflow.com/questions/44835156/docker-for-win-acme-json-permissions). I'll answer both, since this may count as a bug. Similar questions are asked in #1459. 

@JaronrH 
> [Can I change permissions on shared volumes for container-specific deployment requirements?](https://docs.docker.com/docker-for-windows/faqs/#can-i-change-permissions-on-shared-volumes-for-container-specific-deployment-requirements)
>
> No, at this point, Docker for Windows does not enable you to control (chmod) the Unix-style permissions on shared volumes for deployed containers, but rather sets permissions to a default value of 0755 (read, write, execute permissions for user, read and execute for group) which is not configurable.

   Traefik is not compatible with regular Windows due to the POSIX permissions check in #1009 (A bug? A feature?). It may work in the Windows Subsystem for Linux since that has a Unix-style permission system.

An additional cause/solution for the same error message: A Docker volume mount will create a 755 directory if the file does not exist. Create the empty file first.

**Comment by guerillamos** [comment]
Stumbled across this issue when trying to get traefik running on Docker for Windows... ended up getting it working by adding a few lines to a dockerfile to create the acme.json and set permissions. I then built the image and despite throwing the "Docker image from Windows against a non-Windows Docker host security warning" when I checked permissions on the acme.json file it worked!

![chmod600](https://user-images.githubusercontent.com/20997508/45253385-1a357500-b31b-11e8-8060-35e6e3a3371f.png)

I setup a repo up and have it auto building to the dockerhub here for further testing.

[https://hub.docker.com/r/guerillamos/traefik/](url)

Once I got that built I switched the image out in my docker-compose file and my DNS challenge to Cloudflare worked like a charm according to the logs.

I hope this helps someone!
